lib/Notes/NotesSmartAccordion/tests/notes-smart-accordion-test.js

```diff
@@ -0,0 +1,256 @@
+  });
+
+  it('should display notes list', () => {
+    expect(notesAccordion.notesListDisplayed).to.be.true;
+  });
+
+  describe('when new button was clicked', () => {
+    beforeEach(async () => {
+      await notesAccordion.newButton.click();
+    });
+
+    it('should redirect to create note page', function () {
+      expect(this.location.pathname).to.equal('/dummy/notes/new');
+    });
+  });
+
+  describe('when a note in the notes list was clicked', () => {
+    beforeEach(async () => {
+      await notesAccordion.notes(0).click();
+    });
+
+    it('should redirect to note view page', function () {
+      expect(this.location.pathname + this.location.search).to.equal('/dummy/notes/providerNoteId');
+    });
+  });
+
+  describe('when assign button was clicked', () => {
+    beforeEach(async () => {
+      await notesAccordion.assignButton.click();
+    });
+
+    it('should open notes modal', () => {
+      expect(notesModal.isDisplayed).to.be.true;
+    });
+
+    it('should disable search button', () => {
+      expect(notesModal.searchButtonIsDisabled).to.be.true;
+    });
+
+    it('should display empty message', () => {
+      expect(notesModal.emptyMessageIsDisplayed).to.be.true;
+    });
+
+    describe('after click on note assigned to only one entity', () => {
+      beforeEach(async () => {
+        await notesModal.selectAssignedFilter();
+        await notesModal.notes(0).clickCheckbox();
+      });
+
+      it('should not allow to unassign', () => {
+        expect(notesModal.notes(0).checkboxIsDisabled).to.be.true;
+        expect(notesModal.notes(0).checkboxIsSelected).to.be.true;
+      });
+    });
+
+    describe('after click on note assigned to a couple of entities and hitting save', () => {
+      beforeEach(async () => {
+        await notesModal.selectAssignedFilter();
+        await notesModal.notes(1).clickCheckbox();
+        await notesModal.clickSaveButton();
+      });
+
+      it('should unassign it', () => {
+        expect(notesAccordion.notes().length).to.equal(1);
+      });
+    });
+
+    describe('after click on "assign/unassign all" checkbox', () => {
+      beforeEach(async () => {
+        await notesModal.selectAssignedFilter();
+        await notesModal.clickAssignUnassignAllCheckbox();
+      });
+
+      it('should not allow to uncheck notes assigned to only one entity', () => {
+        expect(notesModal.notes(0).checkboxIsSelected).to.be.true;
+      });
+
+      it('should uncheck notes assigned to more then one entity', () => {
+        expect(notesModal.notes(1).checkboxIsSelected).to.be.false;
+      });
+    });
+
+    describe('and search box and filters are not empty', () => {
+      beforeEach(async () => {
+        await notesModal.enterSearchQuery('some note');
+        await wait(300);
+        await notesModal.noteTypeFilter.options(1).clickOption();
+        await notesModal.selectUnassignedFilter();
+        await notesModal.resetAll();
```

Comment:
it seems like what happens in `beforeEach` is different from what is stated in `describe`